cli/client/response.go

```diff
@@ -1,78 +1,63 @@
 package client
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"log"
 	"net/http"
-
-	"github.com/mesosphere/dcos-commons/cli/config"
 )
 
-// Fake functions that allow us to assert against output
-var LogMessage = log.Printf
-var LogMessageAndExit = log.Fatalf
-var PrintMessage = fmt.Printf
-
-func printResponseError(response *http.Response) {
-	LogMessage("HTTP %s Query for %s failed: %s",
-		response.Request.Method, response.Request.URL, response.Status)
-}
+type responseCheck func(response *http.Response, body []byte) error
 
-func printResponseErrorAndExit(response *http.Response) {
-	LogMessageAndExit("HTTP %s Query for %s failed: %s",
-		response.Request.Method, response.Request.URL, response.Status)
-}
+var customCheck responseCheck
 
-func printServiceNameErrorAndExit(response *http.Response) {
-	printResponseError(response)
-	LogMessage("- Did you provide the correct service name? Currently using '%s', specify a different name with '--name=<name>'.", config.ServiceName)
-	LogMessageAndExit("- Was the service recently installed or updated? It may still be initializing, wait a bit and try again.")
+// SetCustomResponseCheck sets a custom responseCheck that
+// can be used for more advanced handling of HTTP responses.
+func SetCustomResponseCheck(check responseCheck) {
```

Comment:
(Some commands require advanced handling of HTTP responses based on error codes and the message contents - e.g. the Plans API, and the Cosmos API.)